gems/smithy-client/spec/smithy-client/codec/cbor_spec.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    module Codec
+      describe CBOR do
+        subject { CBOR.new }
+        let(:string_shape) { Shapes::StringShape.new(id: 'string') }
+
+        let(:list_shape) do
+          shape = Shapes::ListShape.new(id: 'list')
```

Comment:
Don't have to make changes now, but you can consider yielding self on all of the complex shapes so that constructing can be like:
```
Shapes::ListShape.new(id: 'list') do |list|
  list.set_member(...)
end
```

Reply:
Yeah, I do like that and I think we already do that for ServiceShape? 

But generally, wouldn't be useful for the way we generate shapes.rb since we need to generate all shape objects first so they can be referenced when setting members.  (Well unless we changed the approach to do basically a topological sort.  But that seems overkill).

Reply:
Yeah, it's just useful for tests and one-off construction in implementations.

Reply:
Yeah, we currently do that for `ServiceShape` and `OperationShape`. I agree with Alex that it may not be useful since we need to define all the shapes prior to member adding. We could always change this up.